### PR TITLE
Support Julia 1.11+

### PR DESCRIPTION
**Pull request by omus**
Note at this time Julia 1.11.0-beta1 is the latest released version of Julia 1.11 which is why I used `v"1.11-"` in this PR. Note that the "nightly" CI jobs are setup to `continue-on-error` so reviewers will need to view the output of those jobs to confirm it actually passes.

### CODE REVIEW

File: src/runner.jl [hunk 1]
```diff
@@ -380,27 +380,26 @@ function test!(pkg::AbstractString,
...
-            elseif VERSION >= v"1.4.0"
-                test_project_override = test_folder_has_project_file ?
-                    nothing :
+                else
                     gen_target_project(ctx, pkgspec, pkgspec.path, "test")
-                sandbox_args = (sandbox_args..., test_project_override)
+                end
+            else
+                nothing
+            end
```
Comment (omus) [inline src/runner.jl:395]: Diff is hard to read but this part is just dealing with version differences for calling `gen_target_project` independently from `sandbox` args

**Comment by oxinabox** [comment]
> Note that the "nightly" CI jobs are setup to continue-on-error so reviewers will need to view the output of those jobs to confirm it actually passes.

Can we change this?
I would rather they fail visibly, and then we just ignore them when we decide to merge

**Review by oxinabox — APPROVED** [review]
We really do need to finish #89

**Comment by omus** [comment]
To ensure that I didn't break older versions of Julia not normally tested by the CI workflow I added commit https://github.com/JuliaTesting/TestReports.jl/pull/113/commits/cb7a15d515ac529a59a0ecc6609c5af2942b4846 to validate Julia versions 1.2 - 1.10 all pass tests.

**Comment by omus** [comment]
Julia 1.2 is upset about running `Pkg.rm("Example")`:

```
Installed packages by name in primary environment: Error During Test at /home/runner/work/TestReports.jl/TestReports.jl/test/reportgeneration.jl:39
  Got exception outside of a @test
  The following package names could not be resolved:
   * Example (not found in project or manifest)
  Please specify by known `name=uuid`.
  Stacktrace:
   [1] pkgerror(::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Types.jl:112
   [2] #ensure_resolved#107(::Bool, ::typeof(Pkg.Types.ensure_resolved), ::Pkg.Types.EnvCache, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Types.jl:910
   [3] ensure_resolved at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Types.jl:886 [inlined]
   [4] #rm#35(::Pkg.Types.PackageMode, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.rm), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:127
   [5] rm at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:109 [inlined]
   [6] #rm#34 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:106 [inlined]
   [7] rm at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:106 [inlined]
   [8] #rm#33 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:105 [inlined]
   [9] rm at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:105 [inlined]
   [10] #rm#32(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.rm), ::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:104
   [11] rm(::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:104
   [12] top-level scope at /home/runner/work/TestReports.jl/TestReports.jl/test/reportgeneration.jl:45
...
```
I think I'm going to not investigate this issue further.

**Pull request merged**